Mathlib/GameTheory/FairDivision/Discrete/Basic.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2024 Shreyas Srinivas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shreyas Srinivas
+-/
+
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Ring.Basic
+
+
+namespace Mathlib
+
+namespace DiscreteFairDivision
+
+variable {Agents Goods : Type}
+variable [Fintype Agents] [Fintype Goods]
+variable [DecidableEq Agents]
+variable [DecidableEq Goods] -- Indivisible goods only
+
+
+
+structure Alloc where
+  assign : Agents → Finset Goods
+  disjoint : ∀ a b : Agents, a ≠ b → assign a ∩ assign b = ∅
+
+structure Val where
+  val : Agents → Finset Goods → ℝ
+  empty : ∀ a : Agents, val a ∅ = 0
+  pos : ∀ a : Agents, ∀ s : Finset Goods, val a s ≥ 0
+  mono : ∀ a : Agents, ∀ s t : Finset Goods,
+    s ⊆ t → val a s ≤ val a t
+
+
+variable (alloc : @Alloc Agents Goods _)
```

Comment:
This is good evidence that `Agents` and `Goods` should be explicit arguments to `alloc`.